src/components/scenes/PinLoginScene.tsx

```diff
@@ -99,11 +99,15 @@ export function PinLoginScene(props: Props) {
 
   // Runs once at start:
   React.useEffect(() => {
-    if (userInfo != null && biometryType !== 'FaceID') {
+    if (
+      userInfo != null &&
+      displayUsername !== '' &&
+      biometryType !== 'FaceID'
+    ) {
```

Comment:
It looks like `displayUsername === ''` happens because we set `displayUsername` inside a `useEffect`, so there is a delay before it gets set up. It might make more sense to simply get rid of the delay:

```ts
const displayUsername = React.useMemo(() => 
  userInfo.username ??
sprintf(lstrings.guest_account_id_1s, loginId.slice(loginId.length - 3))
, [userInfo])
```

In that case, we have a valid `displayUsername` the instant we have a `userInfo`